modules/mobile/app/models/mobile/v0/adapters/va_appointments.rb

```diff
@@ -77,23 +77,39 @@ def parse(appointments)
 
         private
 
+        # rubocop:disable Metrics/MethodLength
         def build_appointment_model(appointment_hash, facilities)
-          facility_id = sub_non_prod_id!(appointment_hash[:facility_id])
+          facility_id = Mobile::V0::Appointment.toggle_non_prod_id!(
+            appointment_hash[:facility_id]
+          )
           facilities.add(facility_id) if facility_id
+
           details, type = parse_by_appointment_type(appointment_hash)
+          healthcare_service = healthcare_service(details, type)
           start_date_utc = start_date_utc(appointment_hash)
           time_zone = time_zone(facility_id)
+          start_date_local = start_date_utc.in_time_zone(time_zone)
+          status = status(details, type, start_date_utc)
+
+          cancel_id = if booked_va_appointment?(status, type)
+                        Mobile::V0::Contracts::CancelAppointment.encode_cancel_id(
+                          start_date_local: start_date_local,
+                          clinic_id: appointment_hash[:clinic_id],
+                          facility_id: facility_id,
+                          healthcare_service: healthcare_service
+                        )
+                      end
 
           adapted_hash = {
             id: appointment_hash[:id],
             appointment_type: type,
+            cancel_id: cancel_id,
             comment: comment(details, type),
-            clinic_id: appointment_hash[:clinic_id],
             facility_id: facility_id,
             healthcare_service: healthcare_service(details, type),
             location: location(details, type, facility_id),
             minutes_duration: minutes_duration(details, type),
-            start_date_local: start_date_utc.in_time_zone(time_zone),
+            start_date_local: start_date_local,
             start_date_utc: start_date_utc,
             status: status(details, type, start_date_utc),
```

Comment:
Should this just be `status: status,` since status is set above on (the new) line 92?

Reply:
👍 good catch, I'll use the status that was set